core/src/test/scala/integration/kafka/tools/MirrorMakerIntegrationTest.scala

```diff
@@ -92,8 +92,8 @@ class MirrorMakerIntegrationTest extends KafkaServerTestHarness {
           val data = mirrorMakerConsumer.receive()
           data.topic == topic && new String(data.value) == msg
         } catch {
-          // this exception is thrown if no record is returned within a short timeout, so safe to ignore
-          case _: TimeoutException => false
+          // these exceptions are thrown if no records are returned within the 1s timeout, so safe to ignore
+          case _: TimeoutException | _: NoRecordsException => false
```

Comment:
Is `TimeoutException` ever thrown?

Reply:
Also, we should not hardcode the timeout here as it will go stale if we change the MirrorMaker code.

Reply:
Nope, it is not thrown. Nice catch, I hadn't looked into it. `TimeoutException` is thrown on `CommitSync` and on `ConsoleConsumer#receive()`, but not on `MirrorMakerConsumer#receive`. Must be a left over